internal/manifests/service_monitor.go

```diff
@@ -97,6 +98,25 @@ func NewGatewayServiceMonitor(opts Options) *monitoringv1.ServiceMonitor {
 	return sm
 }
 
+// NewIndexGatewayServiceMonitor creates a k8s service monitor for the lokistack-index-gateway component
+func NewIndexGatewayServiceMonitor(opts Options) *monitoringv1.ServiceMonitor {
+	l := ComponentLabels(LabelGatewayComponent, opts.Name)
+
+	serviceMonitorName := serviceMonitorName(IndexGatewayName(opts.Name))
+	serviceName := serviceNameIndexGatewayHTTP(opts.Name)
+	gwEndpoint := serviceMonitorEndpoint(gatewayInternalPortName, serviceName, opts.Namespace, opts.Flags.EnableTLSServiceMonitorConfig)
+
+	sm := newServiceMonitor(opts.Namespace, serviceMonitorName, l, gwEndpoint)
+
+	if opts.Stack.Tenants != nil {
+		if err := configureServiceMonitorForMode(sm, opts.Stack.Tenants.Mode, opts.Flags); err != nil {
+			return sm
+		}
+	}
+
+	return sm
```

Comment:
This is replicating the lokistack gateway form of configuring the service monitor and thus not correct. The lokistack gateway has a special requirement as a multi-container pod. Apply here the pattern as for the querier in `NewQuerierServiceMonitor`.